src/H5VLconnector.h

```diff
@@ -46,6 +46,50 @@
 /* Public Typedefs */
 /*******************/
 
+/* Types for different ways that objects are located in an HDF5 container */
+typedef enum H5VL_loc_type_t {
+    H5VL_OBJECT_BY_SELF,
+    H5VL_OBJECT_BY_NAME,
+    H5VL_OBJECT_BY_IDX,
+    H5VL_OBJECT_BY_TOKEN
+} H5VL_loc_type_t;
+
+typedef struct H5VL_loc_by_name {
+    const char *name;
+    hid_t       lapl_id;
+} H5VL_loc_by_name_t;
+
+typedef struct H5VL_loc_by_idx {
+    const char *    name;
+    H5_index_t      idx_type;
+    H5_iter_order_t order;
+    hsize_t         n;
+    hid_t           lapl_id;
+} H5VL_loc_by_idx_t;
+
+typedef struct H5VL_loc_by_token {
+    H5O_token_t *token;
+} H5VL_loc_by_token_t;
+
+/* Structure to hold parameters for object locations.
+ * Either: BY_SELF, BY_NAME, BY_IDX, BY_TOKEN
+ *
+ * Note: Leave loc_by_token as the first union member so we
+ *       can perform the simplest initialization of the struct
+ *       without raising warnings.
+ *
+ * Note: BY_SELF requires no union members.
+ */
+typedef struct H5VL_loc_params_t {
+    H5I_type_t      obj_type;
+    H5VL_loc_type_t type;
+    union {
+        H5VL_loc_by_token_t loc_by_token;
+        H5VL_loc_by_name_t  loc_by_name;
+        H5VL_loc_by_idx_t   loc_by_idx;
+    } loc_data;
+} H5VL_loc_params_t;
```

Comment:
Just noise - the structs above needed to be moved up before the definition of the below H5VL_link_create_args_t.